packages/backend/src/app/workers/flow-worker/flow-queue.ts

```diff
@@ -70,24 +71,22 @@ export const flowQueue = {
       const jobKey = await client.get(repeatableJobKey(id));
 
       if (jobKey === null) {
-        throw new ActivepiecesError({
-          code: ErrorCode.JOB_REMOVAL_FAILURE,
-          params: {
-            jobId: id,
-          },
-        });
-      }
-
-      const result = await repeatableJobQueue.removeRepeatableByKey(jobKey);
-      await client.del(repeatableJobKey(id));
-
-      if (!result) {
-        throw new ActivepiecesError({
-          code: ErrorCode.JOB_REMOVAL_FAILURE,
-          params: {
-            jobId: id,
-          },
-        });
+        // If the trigger activation failed, don't let the function fail. 
+        // Just ignore the action. Log an error message indicating that the job with key "${jobKey}" couldn't be found, even though it should exist, and proceed to skip the deletion.
+        logger.error(`Couldn't find job ${jobKey}, even though It should exists, skipping delete`);
+      } else {
+
+        const result = await repeatableJobQueue.removeRepeatableByKey(jobKey);
+        await client.del(repeatableJobKey(id));
+
+        if (!result) {
+          throw new ActivepiecesError({
+            code: ErrorCode.JOB_REMOVAL_FAILURE,
+            params: {
+              jobId: id,
+            },
+          });
+        }
```

Comment:
Could this possibly leave the system with a lot of zombie jobs?

Throwing when encountering anomalies is a safe action to ensure the system is not left in an inconsistent state.

Reply:
Let's keep the discussion there: 
https://github.com/activepieces/activepieces/issues/423